src/Search/SearchHandler.php

```diff
@@ -81,14 +81,21 @@ public function search(AdminInterface $admin, $term, $page = 0, $offset = 20)
 
         $datagrid = $admin->getDatagrid();
 
+        $datagridValues = $datagrid->getValues();
+
         $found = false;
         foreach ($datagrid->getFilters() as $filter) {
-            /** @var $filter FilterInterface */
+            /** @var FilterInterface $filter */
+            $formName = $filter->getFormName();
+
             if ($filter->getOption('global_search', false)) {
                 $filter->setOption('case_sensitive', $this->caseSensitive);
                 $filter->setCondition(FilterInterface::CONDITION_OR);
-                $datagrid->setValue($filter->getFormName(), null, $term);
+                $datagrid->setValue($formName, null, $term);
                 $found = true;
+            } elseif (isset($datagridValues[$formName])) {
+                // Reset the default filter parameter value.
+                $datagrid->setValue($formName, null, null);
```

Comment:
IMO, `removeFilter()` could bring a more explicit behavior than setting `null` as value.

Reply:
I totally overlooked that method, thanks!